### DISCUSSION TITLE
Eslint doesn't detect undeclared variables

### DISCUSSION THREAD

**Discussion by awixor** [eslint/eslint#17425] [Q&A]
Hey guys,

Does anyone know why `eslint` is not detecting undeclared variables used like this:

```js
const { data: userData } = useGetUserQuery({ id }, { skip: !isApiV2 });
```
The variable `isApiV2` is not even declared and eslint didn't detect the error.
`eslint` works fine for me, I am only wondering why only in this case doest catch the issue.

**Comment by aladdin-add** [comment]
it can be caught by [no-undef](https://eslint.org/docs/rules/no-undef). have you enable the rule?

**Reply by awixor** [reply]
I am using react and `no-undef` is not working properly for me, becasue it throws errors when using `JSX`, `window`, `NodeJS`, `navigator...`

**Comment by nzakas** [comment]
@awixor there's on way for us to help you without more details. Please try using the [playground](https://eslint.org/play) to reproduce your issue. We need to see your configuration, full example code, and the actual ESLint output.
